### PR TITLE
fix: Fix messages disappearing on reconnect

### PR DESCRIPTION
**Pull request by tsirlucas**
Thanks @nhannah for the nice pair programming session!

# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: package/src/components/Channel/Channel.tsx [hunk 1]
```diff
@@ -927,8 +971,20 @@ const ChannelWithContext = <
 
       channel.state.clearMessages();
       channel.state.addMessagesSorted(finalMessages);
+
       setHasMore(true);
       copyChannelState();
+      await reloadThread();
+
+      if (failedMessages.length) {
```
Comment (nhannah) [inline package/src/components/Channel/Channel.tsx:979]: Only thought is  maybe we should do this above reloadThread so it's not hungup by the await or a failed reload, the failed threadMessages should be added after the reloadThread as they are now.

**Review by nhannah — APPROVED** [review]

File: package/src/components/Channel/Channel.tsx [hunk 2]
```diff
@@ -872,6 +899,7 @@ const ChannelWithContext = <
        * Allow a buffer of 30 new messages, so that MessageList won't move its scroll position,
        * giving smooth user experience.
        */
+
```
Comment (vishalnarkhede) [inline package/src/components/Channel/Channel.tsx:902]:
```suggestion
```

File: package/src/components/Channel/Channel.tsx [hunk 3]
```diff
@@ -882,6 +910,22 @@ const ChannelWithContext = <
...
+        } as unknown as MessageResponse<At, Ch, Co, Me, Re, Us>);
+
+      const failedMessages = messages
+        .filter((message) => message.status === 'failed')
+        .map(parseMessage);
+
+      const failedThreadMessages = thread
+        ? threadMessages.filter((message) => message.status === 'failed').map(parseMessage)
+        : [];
+
```
Comment (vishalnarkhede) [inline package/src/components/Channel/Channel.tsx:928]: This can go after line 944 I guess? Early returns should get priority over setting these values.

**Review by vishalnarkhede — APPROVED** [review]
Few minor comments. But LGTM otherwise!!

**Pull request merged**